test/test_foreach.py

```diff
@@ -12,14 +12,67 @@
 # kernel code paths.
 N_values = [20, 23] if not TEST_WITH_SLOW else [23, 30, 300]
 
+class RegularFuncWrapper:
+
+    def __init__(self, func):
+        self.func = func
+
+    def __call__(self, inputs, **kwargs):
+        return [self.func(*i, **kwargs) for i in zip(*inputs)]
+
+class InplaceRefWrapper:
+
+    def __init__(self, method_name):
+        self.method_name = method_name
+
+    def __call__(self, inputs, **kwargs):
+        return [getattr(i1, self.method_name)(*rest, **kwargs) for i1, *rest in list(zip(*inputs))]
```

Comment:
you should have gotten an attribute on torch.Tensor in `get_foreach_method_names`, and then call it here with `i1` as the first argument, it'll work like method.